src/partition/repart.c

```diff
@@ -72,6 +72,17 @@
 /* LUKS2 takes off 16M of the partition size with its metadata by default */
 #define LUKS2_METADATA_SIZE (16*1024*1024)
 
+#if !HAVE_LIBCRYPTSETUP
+        struct crypt_device {
+
+        };
+        static inline void sym_crypt_free(struct crypt_device* cd) {
+            if (cd)
+                    free(cd);
+        }
+        DEFINE_TRIVIAL_CLEANUP_FUNC(struct crypt_device *, sym_crypt_free);
```

Comment:
just define:

```c
static inline void sym_crypt_freep(struct crypt_device** cd) {};
```
no need to define a function with a body, after all if library support is not built in we cannot have allocated an object and thus don't have to free anything. Also no need to use then DEFINE_TRIVIAL_CLEANUP_FUNC...